manager/manager.go

```diff
@@ -779,21 +777,31 @@ func (m *Manager) rotateRootCAKEK(ctx context.Context, clusterID string) error {
 			return err
 		}
 		unencryptedKeyBlock := &pem.Block{
-			Type:    keyBlock.Type,
-			Bytes:   unencryptedKey,
-			Headers: keyBlock.Headers,
+			Type:  keyBlock.Type,
+			Bytes: unencryptedKey,
 		}
 
-		// We were able to decrypt the key, but with the previous passphrase. Let's encrypt
-		// with the new one and store it in raft
-		finalKey, err = ca.EncryptECPrivateKey(pem.EncodeToMemory(unencryptedKeyBlock), strPassphrase)
+		// we were able to decrypt the key with the previous passphrase - if the current passphrase is empty,
+		// the we store the decrypted key in raft
+		finalKey = pem.EncodeToMemory(unencryptedKeyBlock)
+
+		// the current passphrase is not empty, so let's encrypt with the new one and store it in raft
+		if strPassphrase != "" {
+			finalKey, err = ca.EncryptECPrivateKey(finalKey, strPassphrase)
+			if err != nil {
+				log.G(ctx).Debugf("failed to rotate the key-encrypting-key for the root key material of cluster %s", clusterID)
+				return err
+			}
+		}
+	} else if strPassphrase != "" {
+		// If this key is not encrypted, and the passphrase is not nil, then we have to encrypt it
+		finalKey, err = ca.EncryptECPrivateKey(privKeyPEM, strPassphrase)
 		if err != nil {
-			log.G(ctx).Debugf("failed to rotate the key-encrypting-key for the root key material of cluster %s", clusterID)
 			return err
 		}
 	}
 
-	log.G(ctx).Infof("Re-encrypting the root key material of cluster %s", clusterID)
+	log.G(ctx).Infof("Updating the encryption on the root key material of cluster %s", clusterID)
 	// Let's update the key in the cluster object
 	return s.Update(func(tx store.Tx) error {
 		cluster = store.GetCluster(tx, clusterID)
```

Comment:
I don't know if this is within the scope of this PR, but I see that the code below is blindly overwriting `cluster.RootCA.CAKey`, and I think this will need adjustments to support root rotation.